boards/FSM/Src/App/states/App_AllStates.c

```diff
@@ -28,5 +31,25 @@ void App_AllStatesRunOnTick100Hz(struct StateMachine *const state_machine)
         App_CanRx_BMS_VITALS_SetSignal_HEARTBEAT(can_rx, false);
     }
 
+    // Check for torque plausibility
+    float left_torque_req = App_CanMsgs_dcm_invl_command_message_torque_command_invl_decode(
+        App_CanRx_DCM_INVL_COMMAND_MESSAGE_GetSignal_TORQUE_COMMAND_INVL(can_rx));
+    float right_torque_req = App_CanMsgs_dcm_invr_command_message_torque_command_invr_decode(
+        App_CanRx_DCM_INVR_COMMAND_MESSAGE_GetSignal_TORQUE_COMMAND_INVR(can_rx));
+    float fsm_torque_limit = App_CanTx_GetPeriodicSignal_FSM_TORQUE_LIMIT(can_tx);
```

Comment:
FSM torque limit should be calculated before the fault check, not after. I think this will prevent potential delay false tripping

Reply:
I think comparing signals that are currently on the CAN bus (currently in the CAN TX table) makes more sense since that is what the DCM sees

Reply:
I see, good way to eliminate delay!